abstract_greeter.cpp

```diff
@@ -31,6 +31,10 @@
 #include "lowres_greeter.h"
 #include "nice_greeter.h"
 
+/* ugly fix */
+#undef minor
+#undef major
+
```

Comment:
Why do you need that "ugly fix" ?

Reply:
> Why do you need that "ugly fix" ?

I don't.
For me it was just a hint for someone more familiar with the code, that maybe there's a cleaner solution for that.

I can remove the comment if the fix is otherwise ok for you.